File: TOMIMSV4/Entity/Mantenimientos/Bodega/Bodega/clsBeBodega.vb
Public Class clsBeBodega : Inherits clsBeBodegaBase
    Implements ICloneable
    Implements IDisposable

    Public Property Codigo_barra() As String = ""
    Public Property Nombre_comercial() As String = ""
    Public Property Direccion() As String = ""
    Public Property Telefono() As String = ""
    Public Property Email() As String = ""
    Public Property Encargado() As String = ""
    Public Property Ubic_recepcion() As String = ""
    Public Property Ubic_picking() As String = ""
    Public Property Ubic_despacho() As String = ""
    Public Overloads Property Ubic_merma() As String = ""
    Public Property User_agr() As String = ""
    Public Property Fec_agr() As Date = Date.Now
    Public Property User_mod() As String = ""
    Public Property Fec_mod() As Date = Date.Now
    Public Property Activo() As Boolean = False
    Public Property Coordenada_x() As String = ""
    Public Property Coordenada_y() As String = ""
    Public Property Largo() As Double = 0.0
    Public Property Ancho() As Double = 0.0
    Public Property Alto() As Double = 0.0
    Public Property Reservar_stocks_por_linea() As Boolean = False
    Public Property Rechazar_pedido_por_stock() As Boolean = False
    Public Property IdTipoTransaccion() As String = ""
    Public Property Zoom() As Double = 0.0
    Public Property IdMotivoUbicacionDañadoPicking() As Integer = 0
    Public Property cambio_ubicacion_auto() As Boolean = False
    Public Property codigo_bodega_erp As String = ""
    Public Property Cuenta_Ingreso_Mercancias As String = ""
    Public Property Cuenta_Egreso_Mercancias As String = ""
    Public Property Notificacion_Voz() As Boolean = False
    Public Property Control_Tarifa_Servicios() As Boolean = False
    Public Property Id_Motivo_Ubic_Reabasto() As Integer = 0
    Public Property Es_Bodega_Fiscal As Boolean = False
    Public Property habilitar_ingreso_consolidado As Boolean = False
    Public Shadows Property captura_estiba_ingreso As Boolean = False
    Public Shadows Property captura_pallet_no_estandar As Boolean = False
    Public Property valor_porcentaje_iva As Double = 0
    Public Property control_banderas_cliente As Boolean = False

    ''' <summary>
    ''' #EJC20210728:
    ''' Si este parámetro está activo, se permite la verificación por LP o por código de producto desde la lista de
    ''' pedidos en la pantalla de la HH, sin necesidad de ingresar al pedido.
    ''' </summary>
    ''' <returns></returns>
    Public Property Permitir_Verificacion_Consolidada As Boolean = False
    Public Property IdTamañoEtiquetaUbicacionDefecto As Integer = 0
    Public Property Ubicar_Tarimas_Completas_Reabasto As Boolean = False
    Public Property IdTipoTransaccionSalida As Integer = 0
    Public Property Permitir_Eliminar_Documento_Salida As Boolean = False
    Public Property Eliminar_Documento_Salida As Boolean = False

    ''' <summary>
    ''' '#EJC20220330_CEALSA: Si true, se envía en la HH el IdOperadorBodega para filtrar las tareas de verificación
    ''' </summary>
    ''' <returns></returns>
    Public Property Operador_Picking_Realiza_Verificacion As Boolean = False

    ''' <summary>
    ''' #EJC20220330_CEALSA: Si true, se permite realizar el cambio de ubicación de producto 
    ''' que está reservado en picking pero se actualiza el IdUbicacionTemporal.
    ''' </summary>
    ''' <returns></returns>
    Public Property Permitir_Cambio_Ubic_Producto_Picking As Boolean = False

    ''' <summary>
    ''' #EJC20220701: Cuando se listan los estados en la HH, si true, se muestran tambión los estados "buen estado", segón el check de la tabla.
    ''' </summary>
    ''' <returns></returns>
    Public Property Permitir_Buen_Estado_En_Reemplazo As Boolean = False

    ''' <summary>
    ''' #EJC20220707: En el proceso de reemplazo en la HH, si el parámetro está en false, no considerar el lote para filtrar stock para reemplazar.
    ''' </summary>
    ''' <returns></returns>
    Public Property Restringir_Lote_En_Reemplazo As Boolean = False

    ''' <summary>
    ''' #EJC20220707: En el proceso de reemplazo en la HH, si el parámetro está en false, no considerar la fecha de vencimiento para filtrar stock para reemplazar.
    ''' </summary>
    ''' <returns></returns>
    Public Property Restringir_Vencimiento_En_Reemplazo As Boolean = False

    ''' <summary>
    ''' #CKFK20220717: En el proceso de reabastecimiento manual en la HH se listarón la cantidad de registros parametrizados aquí, si es cero se van a listar todos
    ''' </summary>
    ''' <returns></returns>
    Public Property Top_Reabastecimiento_Manual As Integer = 20

    ''' <summary>
    ''' #CKFK20220721: Permitir o no decimales en las transacciones de ingreso, cambios de ubicación, de estados
    ''' </summary>
    ''' <returns></returns>
    Public Property Permitir_Decimales As Boolean = False

    ''' <summary>
    ''' #CKFK20220209: Ordenar el picking por nombre completo
    ''' </summary>
    ''' <returns></returns>
    Public Property Ordenar_Por_Nombre_Completo As Boolean = False

    ''' <summary>
    ''' #CKFK20220209: Ordenar Picking Descendente
    ''' </summary>
    ''' <returns></returns>
    Public Property Ordenar_Picking_Descendente As Boolean = False


    ''' <summary>
    ''' #GT02032023: Indica cuantos dias de antigüedad puede tener un ticket para validar historico
    ''' </summary>
    ''' <returns></returns>
    Public Property Dias_Limite_Retroactivo As Integer = 0

    '#GT13032023: guarda el horario de ejecución del proceso historico
    Public Property Horario_Ejecucion_Historico As TimeSpan

    '#CKFK20230728 Parámetro para filtrar los pedidos en el despacho.
    Public Property Filtrar_Pedidos_Usuario As Boolean = False
    ''' <summary>
    ''' #EJC202310310846: Indica si la fecha de vencimiento debe ser siempre la misma para un lote (aplica en SAP y NAV)
    ''' </summary>
    ''' <returns></returns>
    Public Property Homologar_Lote_Vencimiento As Boolean = False

    ''' <summary>
    ''' #CKCK20240202 Parámetro que va a permitir escanear la licencia en Picking
    ''' </summary>
    ''' <returns></returns>
    Public Property Escanear_Licencia_Picking As Boolean = False

    Public Property IdTipoEtiquetaLicencia As Integer = 0

    Public Property IdSimbologiaLicencia As Integer = 0

    Public Property Interface_SAP As Boolean = False
    Public Property Restringir_Areas_SAP As Boolean = False

    '#AT20240627 Parámetro que va a permitir realizar implosiones o cu con licencias mixtas
    Public Property Control_Pallet_Mixto As Boolean = False
    ''' <summary>
    ''' #EJC20240904:Se agrega el campo despacho_automático para poder saber si en la verificación, al finalizar, 
    ''' se debe ejecutar el proceso de despacho de forma automática y con valores predeterminados.
    ''' </summary>
    ''' <returns></returns>
    Public Property Despacho_Automatico_HH As Boolean = False
    Public Property Limpiar_Campos As Boolean = False
    Public Property Permitir_Cambio_Ubic_Recepcion As Boolean = False
    Public Property Ruta_CDN As String = ""
    Public Property Rango_Dias_Documentos As Integer = 0
    Public Property Agrupar_Sin_Lic_Veri_No_Cons As Boolean = False
    Public Property Advertir_Mpq_Umbas As Boolean = False

    Sub New()
    End Sub
    Sub New(ByRef IdBodega As Integer, ByVal IdPais As Integer, ByVal IdEmpresa As Integer, ByVal codigo As String, ByVal codigo_barra As String, ByVal nombre As String, ByVal nombre_comercial As String, ByVal direccion As String, ByVal telefono As String, ByVal email As String, ByVal encargado As String, ByVal ubic_recepcion As String, ByVal ubic_picking As String, ByVal ubic_despacho As String, ByVal ubic_merma As String, ByVal user_agr As String, ByVal fec_agr As Date, ByVal user_mod As String, ByVal fec_mod As Date, ByVal activo As Boolean, ByVal coordenada_x As String, ByVal coordenada_y As String, ByVal largo As Double, ByVal ancho As Double, ByVal alto As Double, ByVal reservar_stocks_por_linea As Boolean, ByVal rechazar_pedido_por_stock As Boolean, ByVal IdTipoTransaccion As String, ByVal zoom As Double, ByVal IdMotivoUbicacionDañadoPicking As Integer, ByVal cambio_ubicacion_auto As Boolean, ByVal bloquear_lp_hh As Boolean)
        Me.IdBodega = IdBodega
        Me.IdPais = IdPais
        Me.IdEmpresa = IdEmpresa
        Me.Codigo = codigo
        Me.Codigo_barra = codigo_barra
        Me.Nombre = nombre
        Me.Nombre_comercial = nombre_comercial
        Me.Direccion = direccion
        Me.Telefono = telefono
        Me.Email = email
        Me.Encargado = encargado
        Me.Ubic_recepcion = ubic_recepcion
        Me.Ubic_picking = ubic_picking
        Me.Ubic_despacho = ubic_despacho
        Me.Ubic_merma = ubic_merma
        Me.User_agr = user_agr
        Me.Fec_agr = fec_agr
        Me.User_mod = user_mod
        Me.Fec_mod = fec_mod
        Me.Activo = activo
        Me.Coordenada_x = coordenada_x
        Me.Coordenada_y = coordenada_y
        Me.Largo = largo
        Me.Ancho = ancho
        Me.Alto = alto
        Me.Reservar_stocks_por_linea = reservar_stocks_por_linea
        Me.Rechazar_pedido_por_stock = rechazar_pedido_por_stock
        Me.IdTipoTransaccion = IdTipoTransaccion
        Me.Zoom = zoom
        Me.IdMotivoUbicacionDañadoPicking = IdMotivoUbicacionDañadoPicking
        Me.cambio_ubicacion_auto = cambio_ubicacion_auto
        Me.bloquear_lp_hh = bloquear_lp_hh

    End Sub
    Public Function Clone() As Object Implements ICloneable.Clone
        Return MyBase.MemberwiseClone()
    End Function

#Region "IDisposable Support"
    ''' <summary>
    ''' The disposed value
    ''' </summary>
    Private disposedValue As Boolean ' To detect redundant calls

    ' IDisposable
    ''' <summary>
    ''' Releases unmanaged and - optionally - managed resources.
    ''' </summary>
    ''' <param name="disposing"><c>true</c> to release both managed and unmanaged resources; <c>false</c> to release only unmanaged resources.</param>
    Protected Overridable Sub Dispose(disposing As Boolean)
        If Not disposedValue Then
            If disposing Then
                ' TODO: dispose managed state (managed objects).
            End If

            ' TODO: free unmanaged resources (unmanaged objects) and override Finalize() below.
            ' TODO: set large fields to null.
        End If
        disposedValue = True
    End Sub

    ' TODO: override Finalize() only if Dispose(disposing As Boolean) above has code to free unmanaged resources.
    'Protected Overrides Sub Finalize()
    '    ' Do not change this code.  Put cleanup code in Dispose(disposing As Boolean) above.
    '    Dispose(False)
    '    MyBase.Finalize()
    'End Sub

    ' This code added by Visual Basic to correctly implement the disposable pattern.
    ''' <summary>
    ''' Performs application-defined tasks associated with freeing, releasing, or resetting unmanaged resources.
    ''' </summary>
    Public Sub Dispose() Implements IDisposable.Dispose
        ' Do not change this code.  Put cleanup code in Dispose(disposing As Boolean) above.
        Dispose(True)
        ' TODO: uncomment the following line if Finalize() is overridden above.
        ' GC.SuppressFinalize(Me)
        If Empresa IsNot Nothing Then
            Empresa.Dispose()
            Empresa = Nothing
        End If
    End Sub
#End Region

End Class

<Serializable>
Public Class clsBeBodegaBase

    Public Property IdBodega() As Integer = 0
    Public Property IdPais() As Integer = 0
    Public Property IdEmpresa() As Integer = 0
    Public Property Codigo() As String = ""
    Public Property Nombre() As String = ""

    Public Property bloquear_lp_hh As Boolean = False
    Public Property captura_estiba_ingreso As Boolean = False
    Public Property captura_pallet_no_estandar As Boolean = False
    Public Property priorizar_ubicrec_sobre_ubicest As Boolean = False
    Public Property ubic_merma As String = ""
    Public Property ubic_producto_ne As Integer = 0
    Public Property IdProductoEstadoNE As Integer = 0
    ''' <summary>
    ''' --#EJC20220129: Si true, entonces en el proceso de ubicación se valida que la ubicación destino 
    ''' no tenga el equivalente a un pallet, en base a las tarimas y cajas por cama de la presentación.
    ''' </summary>
    ''' <returns></returns>
    Public Property validar_disponibilidad_ubicaicon_destino As Boolean = False
    ''' <summary>
    ''' #EJC20220224: CEALSA, indica si se muestra la columna zona en picking y consulta de existencias al inicio.
    ''' </summary>
    ''' <returns></returns>
    Public Property Mostrar_Area_En_HH As Boolean = False

    '#AT 20220225 Indica si se debe ESCANEAR/CONFIRMAR el codigo del producto en proceso de picking (CEALSA)
    Public Property confirmar_codigo_en_picking As Boolean = False

    '#EJC20220301: El primer paso al futuro de la gestión automática de operadores.
    Public Property control_operador_ubicacion As Boolean = False

    '#EJC20220314: CEALSA, si true, entonces en el cambio de ubicación, al escanear ónicamente licencia, se coloca automáticamente la ubicación de origen.
    Public Property inferir_origen_en_cambio_ubic As Boolean = False

    ''' <summary>
    ''' #EJC20220510: BYB, si true, no reservar producto vencido.
    ''' </summary>
    ''' <returns></returns>
    Public Property despachar_producto_vencido As Boolean = False

    ''' <summary>
    ''' #EJC20220524: MERCOSAL, nuestra primera parametrización del tipo de pantalla para picking (espero que prevalezca el 3 y no cambie mucho en el tiempo)
    ''' </summary>
    ''' <returns></returns>
    Public Property tipo_pantalla_picking As Integer = 1

    ''' <summary>
    ''' #EJC20220524: MERCOSAL, nuestra primera parametrización del tipo de pantalla para picking (espero que prevalezca el 3 y no cambie mucho en el tiempo)
    ''' </summary>
    ''' <returns></returns>
    Public Property tipo_pantalla_recepcion As Integer = 1

    ''' <summary>
    ''' #EJC20220524: MERCOSAL, nuestra primera parametrización del tipo de pantalla para picking (espero que prevalezca el 3 y no cambie mucho en el tiempo)
    ''' </summary>
    ''' <returns></returns>
    Public Property tipo_pantalla_verificacion As Integer = 1

    ''' <summary>
    ''' #CKFK20220610 Parametrización para mostrar la verificacion consolidada o no
    ''' </summary>
    ''' <returns></returns>
    Public Property Verificacion_Consolidada As Boolean = False

    ''' <summary>
    ''' #GT07072022 Parametrización para indicar giro de empresa como Motriz
    ''' </summary>
    ''' <returns></returns>
    Public Property industria_motriz As Boolean = False

    ''' <summary>
    ''' #EJC20220912_1735: Solicitado por Marcelo MERCOPAN para controlar en los procesos de Reemplazo, 
    ''' con cuantos días máximo puedo reemplazar el producto a partir de su fecha de vencimiento.
    ''' </summary>
    ''' <returns></returns>
    Public Property Dias_Maximo_Vencimiento_Reemplazo As Integer = 0

    ''' <summary>
    ''' #EJC202209121941: Permitir en la pantalla de recepcion, recibir múltiples copias de un producto con los mismos parámetros.
    ''' </summary>
    ''' <returns></returns>
    Public Property Permitir_Repeticiones_En_Ingreso As Boolean = False

    ''' <summary>
    ''' #EJC20221005: En el proceso de carga de inventario inicial, validar si la bodega tiene existencias antes de cargar archivo.
    ''' Carolina...
    ''' </summary>
    ''' <returns></returns>
    Public Property Validar_Existencias_Inv_Ini As Boolean = False

    ''' <summary>
    ''' #EJC202211211022: Determina en la HH si se calcula o no la ubicación sugerida.
    ''' </summary>
    ''' <returns></returns>
    Public Property Calcular_Ubicacion_Sugerida_ML As Boolean = False
    Public Property Permitir_Reemplazo_Picking As Boolean = False
    Public Property Permitir_No_Encontrado_Picking As Boolean = False
    Public Property Permitir_Reemplazo_Verificacion As Boolean = False
    Public Property Permitir_Reemplazo_Picking_Misma_Licencia As Boolean = False
    Public Property Liberar_Stock_Despachos_Parciales As Boolean = False
    Public Property Control_Talla_Color As Boolean = False

End Class

File: WMS.EntityCore/Datos_Maestros/clsBeBodega.cs
using System.ComponentModel;
using System.ComponentModel.DataAnnotations.Schema;

namespace WMS.EntityCore.Datos_Maestros
{
    public class clsBeBodega : ICloneable
    {
        [Column("IdBodega")]
        [DisplayName("IdBodega")]
        public int IdBodega { get; set; } = 0;

        [Column("IdPais")]
        [DisplayName("IdPais")]
        public int IdPais { get; set; } = 0;

        [Column("IdEmpresa")]
        [DisplayName("IdEmpresa")]
        public int IdEmpresa { get; set; } = 0;

        [Column("codigo")]
        [DisplayName("codigo")]
        public string Codigo { get; set; } = "";

        [Column("codigo_barra")]
        [DisplayName("codigo_barra")]
        public string Codigo_barra { get; set; } = "";

        [Column("nombre")]
        [DisplayName("nombre")]
        public string Nombre { get; set; } = "";

        [Column("nombre_comercial")]
        [DisplayName("nombre_comercial")]
        public string Nombre_comercial { get; set; } = "";

        [Column("direccion")]
        [DisplayName("direccion")]
        public string Direccion { get; set; } = "";

        [Column("telefono")]
        [DisplayName("telefono")]
        public string Telefono { get; set; } = "";

        [Column("email")]
        [DisplayName("email")]
        public string Email { get; set; } = "";

        [Column("encargado")]
        [DisplayName("encargado")]
        public string Encargado { get; set; } = "";

        [Column("ubic_recepcion")]
        [DisplayName("ubic_recepcion")]
        public string Ubic_recepcion { get; set; } = "";

        [Column("ubic_picking")]
        [DisplayName("ubic_picking")]
        public string Ubic_picking { get; set; } = "";

        [Column("ubic_despacho")]
        [DisplayName("ubic_despacho")]
        public string Ubic_despacho { get; set; } = "";

        [Column("ubic_merma")]
        [DisplayName("ubic_merma")]
        public string Ubic_merma { get; set; } = "";

        [Column("user_agr")]
        [DisplayName("user_agr")]
        public string User_agr { get; set; } = "";

        [Column("fec_agr")]
        [DisplayName("fec_agr")]
        public DateTime Fec_agr { get; set; } = DateTime.Now;

        [Column("user_mod")]
        [DisplayName("user_mod")]
        public string User_mod { get; set; } = "";

        [Column("fec_mod")]
        [DisplayName("fec_mod")]
        public DateTime Fec_mod { get; set; } = DateTime.Now;

        [Column("activo")]
        [DisplayName("activo")]
        public bool Activo { get; set; } = false;

        [Column("coordenada_x")]
        [DisplayName("coordenada_x")]
        public string Coordenada_x { get; set; } = "";

        [Column("coordenada_y")]
        [DisplayName("coordenada_y")]
        public string Coordenada_y { get; set; } = "";

        [Column("largo")]
        [DisplayName("largo")]
        public double Largo { get; set; } = 0;

        [Column("ancho")]
        [DisplayName("ancho")]
        public double Ancho { get; set; } = 0;

        [Column("alto")]
        [DisplayName("alto")]
        public double Alto { get; set; } = 0;

        [Column("reservar_stocks_por_linea")]
        [DisplayName("reservar_stocks_por_linea")]
        public bool Reservar_stocks_por_linea { get; set; } = false;

        [Column("rechazar_pedido_por_stock")]
        [DisplayName("rechazar_pedido_por_stock")]
        public bool Rechazar_pedido_por_stock { get; set; } = false;

        [Column("IdTipoTransaccion")]
        [DisplayName("IdTipoTransaccion")]
        public string IdTipoTransaccion { get; set; } = "";

        [Column("zoom")]
        [DisplayName("zoom")]
        public double Zoom { get; set; } = 0;

        [Column("IdMotivoUbicacionDañadoPicking")]
        [DisplayName("IdMotivoUbicacionDañadoPicking")]
        public int IdMotivoUbicacionDañadoPicking { get; set; } = 0;

        [Column("cambio_ubicacion_auto")]
        [DisplayName("cambio_ubicacion_auto")]
        public bool Cambio_ubicacion_auto { get; set; } = false;

        [Column("codigo_bodega_erp")]
        [DisplayName("codigo_bodega_erp")]
        public string Codigo_bodega_erp { get; set; } = "";

        [Column("ubic_producto_ne")]
        [DisplayName("ubic_producto_ne")]
        public int Ubic_producto_ne { get; set; } = 0;

        [Column("IdProductoEstadoNE")]
        [DisplayName("IdProductoEstadoNE")]
        public int IdProductoEstadoNE { get; set; } = 0;

        [Column("Cuenta_Ingreso_Mercancias")]
        [DisplayName("Cuenta_Ingreso_Mercancias")]
        public string Cuenta_Ingreso_Mercancias { get; set; } = "";

        [Column("Cuenta_Egreso_Mercancias")]
        [DisplayName("Cuenta_Egreso_Mercancias")]
        public string Cuenta_Egreso_Mercancias { get; set; } = "";

        [Column("notificacion_voz")]
        [DisplayName("notificacion_voz")]
        public bool Notificacion_voz { get; set; } = false;

        [Column("control_tarifa_servicios")]
        [DisplayName("control_tarifa_servicios")]
        public bool Control_tarifa_servicios { get; set; } = false;

        [Column("Id_Motivo_Ubic_Reabasto")]
        [DisplayName("Id_Motivo_Ubic_Reabasto")]
        public int Id_Motivo_Ubic_Reabasto { get; set; } = 0;

        [Column("operador_defecto_en_documento_ingreso")]
        [DisplayName("operador_defecto_en_documento_ingreso")]
        public bool Operador_defecto_en_documento_ingreso { get; set; } = false;

        [Column("es_bodega_fiscal")]
        [DisplayName("es_bodega_fiscal")]
        public bool Es_bodega_fiscal { get; set; } = false;

        [Column("habilitar_ingreso_consolidado")]
        [DisplayName("habilitar_ingreso_consolidado")]
        public bool Habilitar_ingreso_consolidado { get; set; } = false;

        [Column("bloquear_lp_hh")]
        [DisplayName("bloquear_lp_hh")]
        public bool Bloquear_lp_hh { get; set; } = false;

        [Column("captura_estiba_ingreso")]
        [DisplayName("captura_estiba_ingreso")]
        public bool Captura_estiba_ingreso { get; set; } = false;

        [Column("captura_pallet_no_estandar")]
        [DisplayName("captura_pallet_no_estandar")]
        public bool Captura_pallet_no_estandar { get; set; } = false;

        [Column("valor_porcentaje_iva")]
        [DisplayName("valor_porcentaje_iva")]
        public double Valor_porcentaje_iva { get; set; } = 0;

        [Column("Permitir_Verificacion_Consolidada")]
        [DisplayName("Permitir_Verificacion_Consolidada")]
        public bool Permitir_Verificacion_Consolidada { get; set; } = false;

        [Column("control_banderas_cliente")]
        [DisplayName("control_banderas_cliente")]
        public bool Control_banderas_cliente { get; set; } = false;

        [Column("IdTamañoEtiquetaUbicacionDefecto")]
        [DisplayName("IdTamañoEtiquetaUbicacionDefecto")]
        public int IdTamañoEtiquetaUbicacionDefecto { get; set; } = 0;

        [Column("priorizar_ubicrec_sobre_ubicest")]
        [DisplayName("priorizar_ubicrec_sobre_ubicest")]
        public bool Priorizar_ubicrec_sobre_ubicest { get; set; } = false;

        [Column("ubicar_tarimas_completas_reabasto")]
        [DisplayName("ubicar_tarimas_completas_reabasto")]
        public bool Ubicar_tarimas_completas_reabasto { get; set; } = false;

        [Column("validar_disponibilidad_ubicaicon_destino")]
        [DisplayName("validar_disponibilidad_ubicaicon_destino")]
        public bool Validar_disponibilidad_ubicaicon_destino { get; set; } = false;

        [Column("IdTipoTransaccionSalida")]
        [DisplayName("IdTipoTransaccionSalida")]
        public int IdTipoTransaccionSalida { get; set; } = 0;

        [Column("permitir_eliminar_documento_salida")]
        [DisplayName("permitir_eliminar_documento_salida")]
        public bool Permitir_eliminar_documento_salida { get; set; } = false;

        [Column("mostrar_area_en_hh")]
        [DisplayName("mostrar_area_en_hh")]
        public bool Mostrar_area_en_hh { get; set; } = false;

        [Column("confirmar_codigo_en_picking")]
        [DisplayName("confirmar_codigo_en_picking")]
        public bool Confirmar_codigo_en_picking { get; set; } = false;

        [Column("control_operador_ubicacion")]
        [DisplayName("control_operador_ubicacion")]
        public bool Control_operador_ubicacion { get; set; } = false;

        [Column("inferir_origen_en_cambio_ubic")]
        [DisplayName("inferir_origen_en_cambio_ubic")]
        public bool Inferir_origen_en_cambio_ubic { get; set; } = false;

        [Column("eliminar_documento_salida")]
        [DisplayName("eliminar_documento_salida")]
        public bool Eliminar_documento_salida { get; set; } = false;

        [Column("operador_picking_realiza_verificacion")]
        [DisplayName("operador_picking_realiza_verificacion")]
        public bool Operador_picking_realiza_verificacion { get; set; } = false;

        [Column("permitir_cambio_ubic_producto_picking")]
        [DisplayName("permitir_cambio_ubic_producto_picking")]
        public bool Permitir_cambio_ubic_producto_picking { get; set; } = false;

        [Column("despachar_producto_vencido")]
        [DisplayName("despachar_producto_vencido")]
        public bool Despachar_producto_vencido { get; set; } = false;

        [Column("TIPO_PANTALLA_PICKING")]
        [DisplayName("TIPO_PANTALLA_PICKING")]
        public int TIPO_PANTALLA_PICKING { get; set; } = 0;

        [Column("verificacion_consolidada")]
        [DisplayName("verificacion_consolidada")]
        public bool Verificacion_consolidada { get; set; } = false;

        [Column("tipo_pantalla_recepcion")]
        [DisplayName("tipo_pantalla_recepcion")]
        public int Tipo_pantalla_recepcion { get; set; } = 0;

        [Column("tipo_pantalla_verificacion")]
        [DisplayName("tipo_pantalla_verificacion")]
        public int Tipo_pantalla_verificacion { get; set; } = 0;

        [Column("PERMITIR_BUEN_ESTADO_EN_REEMPLAZO")]
        [DisplayName("PERMITIR_BUEN_ESTADO_EN_REEMPLAZO")]
        public bool PERMITIR_BUEN_ESTADO_EN_REEMPLAZO { get; set; } = false;

        [Column("restringir_vencimiento_en_reemplazo")]
        [DisplayName("restringir_vencimiento_en_reemplazo")]
        public bool Restringir_vencimiento_en_reemplazo { get; set; } = false;

        [Column("restringir_lote_en_reemplazo")]
        [DisplayName("restringir_lote_en_reemplazo")]
        public bool Restringir_lote_en_reemplazo { get; set; } = false;

        [Column("industria_motriz")]
        [DisplayName("industria_motriz")]
        public bool Industria_motriz { get; set; } = false;

        [Column("top_reabastecimiento_manual")]
        [DisplayName("top_reabastecimiento_manual")]
        public int Top_reabastecimiento_manual { get; set; } = 0;

        [Column("permitir_decimales")]
        [DisplayName("permitir_decimales")]
        public bool Permitir_decimales { get; set; } = false;

        [Column("permitir_repeticiones_en_ingreso")]
        [DisplayName("permitir_repeticiones_en_ingreso")]
        public bool Permitir_repeticiones_en_ingreso { get; set; } = false;

        [Column("dias_maximo_vencimiento_reemplazo")]
        [DisplayName("dias_maximo_vencimiento_reemplazo")]
        public int Dias_maximo_vencimiento_reemplazo { get; set; } = 0;

        [Column("validar_existencias_inv_ini")]
        [DisplayName("validar_existencias_inv_ini")]
        public bool Validar_existencias_inv_ini { get; set; } = false;

        [Column("calcular_ubicacion_sugerida_ml")]
        [DisplayName("calcular_ubicacion_sugerida_ml")]
        public bool Calcular_ubicacion_sugerida_ml { get; set; } = false;

        [Column("permitir_reemplazo_picking")]
        [DisplayName("permitir_reemplazo_picking")]
        public bool Permitir_reemplazo_picking { get; set; } = false;

        [Column("permitir_no_encontrado_picking")]
        [DisplayName("permitir_no_encontrado_picking")]
        public bool Permitir_no_encontrado_picking { get; set; } = false;

        [Column("permitir_reemplazo_verificacion")]
        [DisplayName("permitir_reemplazo_verificacion")]
        public bool Permitir_reemplazo_verificacion { get; set; } = false;

        [Column("ordenar_por_nombre_completo")]
        [DisplayName("ordenar_por_nombre_completo")]
        public bool Ordenar_por_nombre_completo { get; set; } = false;

        [Column("ordenar_picking_descendente")]
        [DisplayName("ordenar_picking_descendente")]
        public bool Ordenar_picking_descendente { get; set; } = false;

        [Column("Permitir_Reemplazo_Picking_Misma_Licencia")]
        [DisplayName("Permitir_Reemplazo_Picking_Misma_Licencia")]
        public bool Permitir_Reemplazo_Picking_Misma_Licencia { get; set; } = false;

        [Column("dias_limite_retroactivo")]
        [DisplayName("dias_limite_retroactivo")]
        public int Dias_limite_retroactivo { get; set; } = 0;

        [Column("horario_ejecucion_historico")]
        [DisplayName("horario_ejecucion_historico")]
        public DateTime Horario_ejecucion_historico { get; set; } = DateTime.Now;

        [Column("filtrar_pedidos_usuario")]
        [DisplayName("filtrar_pedidos_usuario")]
        public bool Filtrar_pedidos_usuario { get; set; } = false;

        [Column("liberar_stock_depachos_parciales")]
        [DisplayName("liberar_stock_depachos_parciales")]
        public bool Liberar_stock_depachos_parciales { get; set; } = false;

        [Column("liberar_stock_despachos_parciales")]
        [DisplayName("liberar_stock_despachos_parciales")]
        public bool Liberar_stock_despachos_parciales { get; set; } = false;

        [Column("homologar_lote_vencimiento")]
        [DisplayName("homologar_lote_vencimiento")]
        public bool Homologar_lote_vencimiento { get; set; } = false;

        [Column("escanear_licencia_picking")]
        [DisplayName("escanear_licencia_picking")]
        public bool Escanear_licencia_picking { get; set; } = false;

        [Column("interface_SAP")]
        [DisplayName("interface_SAP")]
        public bool Interface_SAP { get; set; } = false;

        [Column("IdTipoEtiquetaLicencia")]
        [DisplayName("IdTipoEtiquetaLicencia")]
        public int IdTipoEtiquetaLicencia { get; set; } = 0;

        [Column("IdSimbologiaLicencia")]
        [DisplayName("IdSimbologiaLicencia")]
        public int IdSimbologiaLicencia { get; set; } = 0;

        [Column("restringir_areas_sap")]
        [DisplayName("restringir_areas_sap")]
        public bool Restringir_areas_sap { get; set; } = false;

        [Column("control_pallet_mixto")]
        [DisplayName("control_pallet_mixto")]
        public bool Control_pallet_mixto { get; set; } = false;

        [Column("despacho_automatico_hh")]
        [DisplayName("despacho_automatico_hh")]
        public bool Despacho_automatico_hh { get; set; } = false;
        public bool Control_Talla_Color { get; set; } = false;
        public bool Reservar_primero_almacenaje { get; set; } = false;
        public clsBeBodega() { }

        public object Clone()
        {
            return MemberwiseClone();
        }
    }
}